contrib/IECoreUSD/src/IECoreUSD/PrimitiveAlgo.cpp

```diff
@@ -239,13 +245,111 @@ void readPrimitiveVariable( const pxr::UsdGeomPrimvar &primVar, pxr::UsdTimeCode
+	// and be used across stages, so we use a global cache for the process.
+	static pxr::UsdSkelCache *g_skelCache = nullptr;
+	if( !g_skelCache )
+	{
+		g_skelCache = new pxr::UsdSkelCache;
+	}
+
+	return g_skelCache;
+}
+
+bool readPrimitiveVariables( const pxr::UsdSkelRoot &skelRoot, const pxr::UsdGeomPointBased &pointBased, pxr::UsdTimeCode time, IECoreScene::Primitive *primitive )
+{
+	pxr::UsdSkelSkeletonQuery skelQuery = ::skelCache()->GetSkelQuery( pxr::UsdSkelBindingAPI( pointBased.GetPrim() ).GetInheritedSkeleton() );
+	if( !skelQuery )
+	{
+		return false;
+	}
+
+	pxr::VtMatrix4dArray skinningXforms;
+	if( !skelQuery.ComputeSkinningTransforms( &skinningXforms, time ) )
+	{
+		return false;
+	}
+
+#if USD_VERSION < 2011
+	::skelCache()->Populate( skelRoot );
+#else
+	::skelCache()->Populate( skelRoot, pxr::UsdTraverseInstanceProxies() );
+#endif
+
+	pxr::UsdSkelSkinningQuery skinningQuery = ::skelCache()->GetSkinningQuery( pointBased.GetPrim() );
+	if( !skinningQuery )
+	{
+		return false;
+	}
+
+	pxr::VtVec3fArray points;
+	if( !pointBased.GetPointsAttr().Get( &points, time ) )
+	{
+		return false;
+	}
+
+	if( !skinningQuery.ComputeSkinnedPoints( skinningXforms, &points, time ) )
+	{
+		return false;
+	}
+
+	auto p = boost::static_pointer_cast<V3fVectorData>( DataAlgo::fromUSD( points ) );
+	if( !p )
+	{
+		return false;
+	}
+
+	p->setInterpretation( GeometricData::Point );
+	primitive->variables["P"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, p );
+
+	// we'll consider normals optional and return true regardless of whether normals were skinned successfully
+	pxr::VtVec3fArray normals;
+	if( pointBased.GetNormalsAttr().Get( &normals, time ) && skinningQuery.ComputeSkinnedNormals( skinningXforms, &normals, time ) )
+	{
+		if( auto n = boost::static_pointer_cast<V3fVectorData>( DataAlgo::fromUSD( normals ) ) )
+		{
+			n->setInterpretation( GeometricData::Normal );
+			primitive->variables["N"] = IECoreScene::PrimitiveVariable( PrimitiveAlgo::fromUSD( pointBased.GetNormalsInterpolation() ), n );
+		}
+	}
+
+	return true;
+}
+
+bool skelAnimMightBeTimeVarying( const pxr::UsdPrim &prim )
+{
+	pxr::UsdSkelSkeletonQuery skelQuery = ::skelCache()->GetSkelQuery( pxr::UsdSkelBindingAPI( prim ).GetInheritedSkeleton() );
+	if( !skelQuery )
+	{
+		return false;
+	}
+
+	const pxr::UsdSkelAnimQuery &animQuery = skelQuery.GetAnimQuery();
+	if( !animQuery )
+	{
+		return false;
+	}
+
+	return animQuery.JointTransformsMightBeTimeVarying();
+}
+
 } // namespace
 
 void IECoreUSD::PrimitiveAlgo::readPrimitiveVariables( const pxr::UsdGeomPrimvarsAPI &primvarsAPI, pxr::UsdTimeCode time, IECoreScene::Primitive *primitive )
 {
 	for( const auto &primVar : primvarsAPI.GetPrimvars() )
 	{
 		string name = primVar.GetPrimvarName().GetString();
+
+		// ignore the UsdSkel primvars as they are not valid Cortex PrimitiveVariables
+		// for this primitive (ie. they refer to the skeleton, not the mesh)
```

Comment:
I guess my comment was trying to say that a non-default `elementSize` in Cortex means your `PrimitiveVariable` is invalid. If I don't skip these, the mesh fails to draw and the message in the terminal is "Mesh with invalid primitive variables" (because `mesh->arePrimitiveVariablesValid()` returns false.

Reply:
I found "they refer to the skeleton, not the mesh" fairly confusing as someone coming to it cold. Would be great if we could clarify the comment a bit.

Reply:
I took a shot at clarifying in 94450bd3ab5b11f13721e82db2d01102845d0f1a